src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/BrokeredMessageToStringConverter.cs

```diff
@@ -2,54 +2,76 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Globalization;
 using System.IO;
+using System.Runtime.Serialization;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml;
 using Microsoft.Azure.WebJobs.Host.Converters;
 using Microsoft.ServiceBus.Messaging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Triggers
 {
     internal class BrokeredMessageToStringConverter : IAsyncConverter<BrokeredMessage, string>
     {
-        public Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
+        public async Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
         {
             if (input == null)
             {
                 throw new ArgumentNullException("input");
             }
+            Stream stream = input.GetBody<Stream>();
+            if (stream == null)
+            {
+                return null;
+            }
 
-            if (input.ContentType == ContentTypes.TextPlain ||
```

Comment:
I agree it's not obvious. I had to write some tests to make sure I understood (there's combinations of all this in the tests I added). There's (at least) two ways you can create a BrokeredMessage and pass it bytes:
1. You pass bytes in and let the BrokeredMessage deal with the serialization:
     
        byte[] bytes = Encoding.UTF8.GetBytes("test");
        BrokeredMessage message = new BrokeredMessage(bytes);
2. You create a Stream yourself and pass that in: 

        MemoryStream ms = new MemoryStream();
        StreamWriter sw = new StreamWriter(ms);
        sw.Write("test");
        sw.Flush();
        ms.Seek(0, SeekOrigin.Begin);
        BrokeredMessage message = new BrokeredMessage(ms);

And then there are two ways to try to get the bytes out:
A. `message.GetBody<byte[]>()`
B. `message.GetBody<Stream>()` and then read as a `MemoryStream`

Let me run through what happens if you mix these and then read the bytes back into a string:
1A: `test`
1B: `@base64Binary3http://schemas.microsoft.com/2003/10/Serialization/�test`
2A: `SerializationException: The input source is not correctly formatted.`
2B: `test`

So if you're writing in as a Stream, you need to tell us via the `ContentType` so we can read it out correctly and give you 2B -- otherwise you end up with the exception in 2A.
I imagine the most common path is 1A, but the other support was always there so I didn't do much to touch it, other than provide a nicer message should you hit 2A.